### ISSUE TITLE
List not showing enough elements when big differences in height

### ISSUE THREAD

**Issue by Shadowlauch** [petyosi/react-virtuoso#60]
This is a bit of a special and maybe even contrived case, at least to the extent I did it, but this could lead to mistakes in more reasonable examples aswell.

Virtuoso doesn't really handle big differences in height very well, when there is a sudden decrease in size.

```
<Virtuoso style={{ width: '100%', height: '400px' }} overscan={400} totalCount={200} item={(index: number) => {
            return <div style={{height: (index < 100 ? 200 : 20)}}>test {index}</div>;
        }} />
```

Since the first 100 items have a height of 200 Virtuoso assumes that height for all elements to begin with and once you scroll down to the items with a height of 20 it does not render enough items to fill the viewport. 

I'm not sure if I'm doing something wrong or this is simply a limitation of the list.

**Comment by petyosi** [comment]
You are correct in your statement - the list assumes the "default" value from the first item rendered (it has to take it from somewhere, and maybe a "defaultItemHeight" property would help). 

However, there is a self-repairing mechanism that will append additional items if the assumption is wrong. I took your example and it seems to [work fine here](https://stackblitz.com/edit/react-compvw?file=example.js), but perhaps I am missing something.

Can you please elaborate on how I can reproduce the incorrect behavior.

**Comment by Shadowlauch** [comment]
To reproduce the error I got you have to scroll fast down to the smaller items and then abruptly stop. That sometimes leads to an empty viewport. It is a bit easier to do with a larger list size: https://stackblitz.com/edit/react-compvw-bkco4s

Once you are in that state the list does not seem to use the recovery mechanism you spoke of anymore.
![2020-01-09-14-04-07](https://user-images.githubusercontent.com/923322/72070260-64e94780-32e9-11ea-8721-098df3a6fa1c.gif)

**Comment by petyosi** [comment]
Got it, thanks for the repro steps!

**Comment by petyosi** [comment]
Confirming that this is a bug, I found a solution but need to test it for possible performance regression. If everything is good, I will release a fix over the weekend.

**Comment by Shadowlauch** [comment]
@petyosi Thanks for the quick response!

**Comment by petyosi** [comment]
Published 0.12.4 and [updated the example](https://stackblitz.com/edit/react-compvw-jqanm6?file=package.json) with it. You can see how the shorter items are re-filled with new ones.

A property that can improve the behavior in this case is `defaultItemHeight` - something which I consider.
